tests/utils/test_qc_check.py

```diff
@@ -0,0 +1,82 @@
+from BALSAMIC.utils.qc_check import read_hs_metrics, read_qc_table, check_qc_criteria, write_output
+from BALSAMIC.utils.qc_check import get_bait_name, get_sample_name, get_qc_criteria, failed_qc
+from BALSAMIC.utils.constants import HSMETRICS_QC_CHECK
+import os
+
+
+def test_read_hs_metrics():
+    # GIVEN the file exist
+    hs_metrics_path = "tests/test_data/qc_files/multiqc_picard_HsMetrics.json"
+
+    # WHEN reading the file
+    df = read_hs_metrics(hs_metrics_path)
+
+    # THEN check if the file contains any values
+    bol_list = df.any().tolist()
+    for n in range(len(bol_list)):
+        assert bol_list[n], "No values exists"
+
+
+def test_read_qc_table():
+    # GIVEN the file exist
+    try:
+        open("/BALSAMIC/utils/constants.py")
+    except IOError:
+        print("File doesn't exist")
+
+    # WHEN reading the file
+    df = read_qc_table(HSMETRICS_QC_CHECK)
+
+    # THEN check if the file contains any values
+    bol_list = df.any().tolist()
+    for n in range(len(bol_list)):
+        assert bol_list[n], "No values exists"
+
+
+def test_get_bait_and_sample_name():
+    # GIVEN the file exists
+    config_file = "tests/test_data/qc_files/config_paired.json"
+
+    # WHEN reading the file
+    bed = get_bait_name(config_file)
+    sample_name = get_sample_name(config_file)
+
+    # THEN check if bed is string format and if sample name exists
+    assert isinstance(bed, str), "bed is not in string format"
+    assert sample_name[0], "sample name doesn't exist"
+
+
+def test_get_qc_criteria():
+    # GIVEN following df and bed
+    df_qc = read_qc_table(HSMETRICS_QC_CHECK)
+    bed = "gmcksolid_4.1_hg19_design.bed"
+
+    # WHEN reading the function
+    criteria_df = get_qc_criteria(df_qc, bed)
+
+    # THEN check if df has two columns
+    nr_of_columns = list(criteria_df.columns)
+    assert len(nr_of_columns) == 2, "number of columns != 2"
+
+
+def test_check_qc_criteria_output_csv_and_qc():
+    # GIVEN following variables
+    config_file = "tests/test_data/qc_files/config_paired.json"
+    hs_metrics = "tests/test_data/qc_files/multiqc_picard_HsMetrics.json"
+    output_path = "tests/test_data/qc_files/output.csv"
```

Comment:
Use PyTest's temporary path factory here:
```suggestion
def test_check_qc_criteria_output_csv_and_qc(tmp_path):
    # GIVEN following variables
    config_file = "tests/test_data/qc_files/config_paired.json"
    hs_metrics = "tests/test_data/qc_files/multiqc_picard_HsMetrics.json"
    
    test_new_dir = tmp_path / "check_qc_results" / "output.csv"
    test_new_dir.mkdir()
    output_path = test_new_dir / "config.json"
```

then `output_path.as_posix()` or `str(output_path)` is the one to use below in assert.

Reply:
Where is `tmp_path` provided when running the pytest?

Reply:
`tmp_path` is also a fixture, this is provided by pytest. very handy to create temporary path and files.

Reply:
Why do you use "config.json" in `output_path`? The `output_path` variable is used in the `write_output` function and should not be config-file, it should just indicate where to put the output.csv file

Reply:
My mistake, I meant "output.csv" or something similar.

So it should be:

```python
test_new_dir = tmp_path / "check_qc_results"
test_new_dir.mkdir()
output_path = Path(test_new_dir / "output.csv").as_posix()
```

Reply:
tmp_path and `mkdir()` build the temporary directory and return a `Path` object. `write_output` doesn't support `Path`objects, that's why I added `as_posix()`. One can use `str()` as well.